lib/cn/hblock_builder.h

```diff
@@ -42,17 +43,25 @@ hbb_destroy(struct hblock_builder *bld);
 merr_t
 hbb_finish(
     struct hblock_builder *bld,
-    struct kvs_block *blkl,
-    uint64_t min_seqno,
-    uint64_t max_seqno,
-    uint32_t num_kblocks,
-    uint32_t num_vblocks,
-    uint32_t num_vgroups,
-    const uint8_t *hlog);
+    struct kvs_block      *blk,
+    const struct vgmap    *vgmap,
+    struct key_obj        *min_pfxp,
+    struct key_obj        *max_pfxp,
+    const uint64_t         min_seqno,
+    const uint64_t         max_seqno,
+    const uint32_t         num_kblocks,
+    const uint32_t         num_vblocks,
+    const uint32_t         num_ptombs,
```

Comment:
Why does this need to be passed by the API when the builder already knows it?

Reply:
Will do that in my next PR.